src/main/java/com/ibm/crypto/plus/provider/XDHPrivateKeyImpl.java

```diff
@@ -57,7 +58,12 @@ final class XDHPrivateKeyImpl extends PKCS8Key implements XECPrivateKey, Seriali
     private void setFieldsFromXeckey() throws Exception {
         if (this.key == null) {
             this.key = extractPrivateKeyFromOCK(xecKey.getPrivateKeyBytes()); // Extract key from GSKit and sets params
-            this.scalar = Optional.of(key);
+            if (Integer.parseInt(provider.getVersionStr()) <= 11) {
+                super.key = this.key;
+            } else {
+                super.key = new DerValue(DerValue.tag_OctetString, this.key).toByteArray();
```

Comment:
I find the existence of a field with the same name, but private, a bit confusing and potentially error prone. Since it is not accessible, maybe we could just use the one variable.

I'm thinking something like:
```
if (Integer.parseInt(provider.getVersionStr()) > 11) {
       this.key = new DerValue(DerValue.tag_OctetString, this.key).toByteArray();
}
```

And I think we're good to set the scalar to `this.key` too.

Same thing goes for other spots too.

Reply:
Before my changes, this class doesn't have a `key` variable. So, when assign the value to `this.key`, it actually assign the value to `PKCS8Key` class `key` object. This caused the private key encoding format issue, because after jdk11 and after version, the private key object in `PKCS8Key` class requires the `[octet string]` before it.

So, I added a variable `private byte[] key;`. All the previous operations on the `this.key` will only work for this `XDHPrivateKeyImpl` class private variable `private byte[] key;`. It won't assign the value to its parent class `PKCS8Key` private `key` object. So, it won't affect the current behaviours in `XDHPrivateKeyImpl` class.

But, we do need to set the `PKCS8Key` private `key` object, for example the encoding purpose. So, I am thinking, to use the `super.key` to explicitly set its parent class `PKCS8Key` private `key` object. And because this `super.key` object is in the `PKCS8Key` class, when the jdk version is 17 or after version, we need to add an `[octet string]` before it.

Reply:
What I don't get is why not assign the `key` object in the `XDHPrivateKeyImpl` class too, after we have fixed the encoding of course. Shouldn't the superclass key also hold the proper encoding in that case too, based always on the Java version?

In other words, why keep `this.key` different than `super.key` since the former one is not reused anywhere anyway. Unless of course I'm missing something in the logic.

Reply:
If we need both, we can always have distinct names, like `rawKey` and `encodedKey`. I just feel like `this.key` and `super.key` is going to be very confusing to anyone reading it in the future.

Reply:
According to our discussion, changed `this.key` to `k` as the local private variable, same as SunEC. Changed `super.key` to `this.key` as the `key` variable in its parent class `PKCS8Key`, the similar way as what SunEC did.